### ISSUE TITLE
Why accuracy is remaining 0?

### ISSUE THREAD

**Issue by RandyCHz** [Yochengliu/Relation-Shape-CNN#7]
Hi, I follow your README to run the code, but the classification accuracy is remaining 0 after several epoch. Is there something important I neglected? Thank you very much!
```
[epoch   4: 160/307] 	 train loss: 0.928064 	 lr: 0.00100
[epoch   4: 180/307] 	 train loss: 0.486494 	 lr: 0.00100
[epoch   4: 200/307] 	 train loss: 0.455191 	 lr: 0.00100
[epoch   4: 220/307] 	 train loss: 0.509143 	 lr: 0.00100
[epoch   4: 240/307] 	 train loss: 0.903722 	 lr: 0.00100
[epoch   4: 260/307] 	 train loss: 0.333634 	 lr: 0.00100
[epoch   4: 280/307] 	 train loss: 0.712478 	 lr: 0.00100

val loss: 0.488999 	 acc: 0.000000

[epoch   4: 300/307] 	 train loss: 0.461941 	 lr: 0.00100
[epoch   5:   0/307] 	 train loss: 0.480295 	 lr: 0.00100
[epoch   5:  20/307] 	 train loss: 0.308337 	 lr: 0.00100
[epoch   5:  40/307] 	 train loss: 0.756740 	 lr: 0.00100
[epoch   5:  60/307] 	 train loss: 0.552190 	 lr: 0.00100
[epoch   5:  80/307] 	 train loss: 0.518123 	 lr: 0.00100
[epoch   5: 100/307] 	 train loss: 0.719212 	 lr: 0.00100
[epoch   5: 120/307] 	 train loss: 0.408349 	 lr: 0.00100
[epoch   5: 140/307] 	 train loss: 0.633257 	 lr: 0.00100

val loss: 0.446945 	 acc: 0.000000

[epoch   5: 160/307] 	 train loss: 0.779942 	 lr: 0.00100
[epoch   5: 180/307] 	 train loss: 0.585827 	 lr: 0.00100
[epoch   5: 200/307] 	 train loss: 0.550597 	 lr: 0.00100
[epoch   5: 220/307] 	 train loss: 0.406125 	 lr: 0.00100
[epoch   5: 240/307] 	 train loss: 0.290858 	 lr: 0.00100
[epoch   5: 260/307] 	 train loss: 0.365490 	 lr: 0.00100
[epoch   5: 280/307] 	 train loss: 0.821365 	 lr: 0.00100

val loss: 0.458882 	 acc: 0.000000

```

**Comment by thtssqc** [comment]
How do you run the code? I followed the README to run ,but when I "make" ,it will appear a error that say "No Model namede torch" ,but I have installed torch==0.4.1 ,so I think the reason is compile to .so not success .Do you have this problem?

**Comment by RandyCHz** [comment]
> How do you run the code? I followed the README to run ,but when I "make" ,it will appear a error that say "No Model namede torch" ,but I have installed torch==0.4.1 ,so I think the reason is compile to .so not success .Do you have this problem?

maybe it is because you haven't activated the torch environment when you used the command `cmake .. && make`

**Comment by Yochengliu** [comment]
@RandyCHz  What version of PyTorch do you use? The val loss looks like all rihgt, you may need to check line 154-159 in train_cls.py, just print them to see what wrong is.

**Comment by Yochengliu** [comment]
@thtssqc   You can run into python environment by typing "python" or "ipython", then type "import torch" to check it. If raising "No Model namede torch" again, maybe you need to check your python environment.

**Comment by RandyCHz** [comment]
> @RandyCHz What version of PyTorch do you use? The val loss looks like all rihgt, you may need to check line 154-159 in train_cls.py, just print them to see what wrong is.

It seems to be a datatype question, I change it to float and it works. Thanks!

**Comment by ANGELALIGN-BOWEN** [comment]
for tensor having only one value,  using tensor.item() instead.
(preds ==labels).sum().item()

**Comment by LX-meteor** [comment]
> > @RandyCHz What version of PyTorch do you use? The val loss looks like all rihgt, you may need to check line 154-159 in train_cls.py, just print them to see what wrong is.
> 
> It seems to be a datatype question, I change it to float and it works. Thanks!

Hi, I got the same problem when I ran the code.Where can I change the data type? How did you solve it?

**Comment by jofury** [comment]
> > > @RandyCHz What version of PyTorch do you use? The val loss looks like all rihgt, you may need to check line 154-159 in train_cls.py, just print them to see what wrong is.
> > 
> > 
> > It seems to be a datatype question, I change it to float and it works. Thanks!
> 
> Hi, I got the same problem when I ran the code.Where can I change the data type? How did you solve it?

At line 159, change (preds ==labels).sum() to (preds ==labels).sum().item() as suggested by @ANGELALIGN-BOWEN

**Comment by LX-meteor** [comment]
@jofury  It works !  Thank you !
